client/src/containers/AddressPage.tsx

```diff
@@ -1,35 +1,64 @@
 import React, { Component } from "react";
 import FileSaver from "file-saver";
-import Browser from "util/browser";
+import Browser from "../util/browser";
 
 import _find from "lodash/find";
 
-import AddressToolbar from "components/AddressToolbar";
-import PropertiesMap from "components/PropertiesMap";
-import PropertiesList from "components/PropertiesList";
-import PropertiesSummary from "components/PropertiesSummary";
-import Indicators from "components/Indicators";
-import DetailView from "components/DetailView";
-import APIClient from "components/APIClient";
-import Loader from "components/Loader";
+import AddressToolbar from "../components/AddressToolbar";
+import PropertiesMap from "../components/PropertiesMap";
+import PropertiesList from "../components/PropertiesList";
+import PropertiesSummary from "../components/PropertiesSummary";
+import Indicators from "../components/Indicators";
+import DetailView from "../components/DetailView";
+import APIClient from "../components/APIClient";
+import Loader from "../components/Loader";
 
 import "styles/AddressPage.css";
 import NychaPage from "./NychaPage";
 import NotRegisteredPage from "./NotRegisteredPage";
 import helpers from "../util/helpers";
 import { Trans, Plural } from "@lingui/macro";
-import { Link } from "react-router-dom";
+import { Link, RouteComponentProps } from "react-router-dom";
 import Page from "../components/Page";
+import { SearchResults, AddressRecord, GeoSearchData } from "../components/APIDataTypes";
 
-export default class AddressPage extends Component {
-  constructor(props) {
+type RouteParams = {
+  locale?: string;
+  boro?: string;
+  housenumber?: string;
+  streetname?: string;
+};
+
+type RouteState = {
+  // TODO: Fix this typing.
+  results?: any;
+};
+
+type AddressPageProps = RouteComponentProps<RouteParams, {}, RouteState> & {
+  currentTab: number;
+};
+
+type State = {
+  hasSearched: boolean;
+  detailMobileSlide: boolean;
+  assocAddrs: AddressRecord[];
+  geosearch?: GeoSearchData;
+
+  // TODO: Fix these typings.
+  searchAddress: any;
+  userAddr: any;
+  detailAddr: any;
+};
+
+export default class AddressPage extends Component<AddressPageProps, State> {
+  constructor(props: AddressPageProps) {
     super(props);
 
     this.state = {
       searchAddress: { ...props.match.params }, // maybe this should be
       userAddr: {}, // merged together?
       hasSearched: false,
-      geosearch: {},
```

Comment:
I'm a bit queasy about this change but hopefully it should work.  We should probably kick the tires a bit just to make sure nothing explodes...